front/src/common/Pathfinding/TypeAndPath.tsx

```diff
@@ -145,43 +196,69 @@ export default function TypeAndPath({ zoomToFeature }: PathfindingProps) {
+  }, [cursorIndex]);
+
   return (
-    <div
-      className="type-and-path"
-      style={{ minWidth: `${monospaceOneCharREMWidth * inputText.length + 5.5}rem` }} // To grow input field & whole div along text size
-      data-testid="type-and-path-container"
-    >
-      <div className="help">{opList.length === 0 && t('inputOPTrigrams')}</div>
-      <OpTooltips opList={opList} />
-      <div className="d-flex align-items-center">
-        <div
-          className={cx('form-control-container', 'flex-grow-1', 'mr-2', {
-            'is-invalid': isInvalid,
-          })}
-        >
-          <input
-            className="form-control form-control-sm text-zone"
-            type="text"
-            value={inputText}
-            onChange={(e) => handleInput(e.target.value)}
-            placeholder={t('inputOPTrigramsExample')}
-            autoFocus
-            data-testid="type-and-path-input"
-          />
-          <span className="form-control-state" />
+    <>
+      <div
+        className="type-and-path mb-2"
+        style={{ minWidth: `${monospaceOneCharREMWidth * inputText.length + 5.5}rem` }} // To grow input field & whole div along text size
+        data-testid="type-and-path-container"
+      >
+        <div className="help">{opList.length === 0 && t('inputOPTrigrams')}</div>
+        <OpTooltips opList={opList} />
+        <div className="d-flex align-items-center">
+          <div
+            className={cx('form-control-container', 'flex-grow-1', 'mr-2', {
+              'is-invalid': isInvalid,
+            })}
+          >
+            <input
+              ref={inputRef}
+              className="form-control form-control-sm text-zone"
+              type="text"
+              value={inputText}
+              onChange={(e) => handleInput(e.target.value)}
+              placeholder={t('inputOPTrigramsExample')}
+              autoFocus
+              data-testid="type-and-path-input"
+            />
+            <span className="form-control-state" />
+          </div>
+          <button
+            className="btn btn-sm btn-success"
+            type="button"
+            aria-label={t('launchPathFinding')}
+            title={t('launchPathFinding')}
+            onClick={launchPathFinding}
+            disabled={isInvalid || opList.length < 2}
+            data-testid="submit-search-by-trigram"
+          >
+            <TriangleRight />
+          </button>
         </div>
-        <button
-          className="btn btn-sm btn-success"
-          type="button"
-          aria-label={t('launchPathFinding')}
-          title={t('launchPathFinding')}
-          onClick={launchPathFinding}
-          disabled={isInvalid || opList.length < 2}
-          data-testid="submit-search-by-trigram"
-        >
-          <TriangleRight />
-        </button>
       </div>
-    </div>
+      {searchResults.length > 0 && <span className="arrow-img"> </span>}
+      {searchResults.length > 0 && (
```

Comment:
Can you optimize this because you use the same condition twice

Reply:
done